dotCMS/src/main/java/com/dotcms/util/RelationshipUtil.java

```diff
@@ -0,0 +1,132 @@
+package com.dotcms.util;
+
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.IdentifierAPI;
+import com.dotmarketing.business.RelationshipAPI;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.languagesmanager.business.LanguageAPI;
+import com.dotmarketing.portlets.structure.model.Field;
+import com.dotmarketing.portlets.structure.model.Relationship;
+import com.dotmarketing.portlets.structure.model.Structure;
+import com.dotmarketing.util.InodeUtils;
+import com.dotmarketing.util.UUIDUtil;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.model.User;
+import com.liferay.util.StringPool;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Utility class used for operations over relationships
+ * @author nollymar
+ */
+public class RelationshipUtil {
+
+    private final static ContentletAPI contentletAPI = APILocator.getContentletAPI();
+
+    private static final IdentifierAPI identifierAPI = APILocator.getIdentifierAPI();
+
+    private static final LanguageAPI languageAPI     = APILocator.getLanguageAPI();
+
+    private final static RelationshipAPI relationshipAPI = APILocator.getRelationshipAPI();
+
+    public static Relationship getRelationshipFromField(final Field field, final Contentlet contentlet){
+        final String fieldRelationType = field.getFieldRelationType();
+        return APILocator.getRelationshipAPI().byTypeValue(
+                fieldRelationType.contains(StringPool.PERIOD) ? fieldRelationType
+                        : contentlet.getContentType().variable() + StringPool.PERIOD + field
+                                .getVelocityVarName());
+
+
+    }
+
+    public static Map<Relationship, List<Contentlet>> getRelatedContentFromQuery(
+            final Relationship relationship, final Contentlet contentlet, final String value,
+            final User user) throws DotDataException, DotSecurityException {
+        List<Contentlet> relatedContentlets = Collections.EMPTY_LIST;
+        if (relationship != null && InodeUtils.isSet(relationship.getInode())) {
+
+            if (UtilMethods.isSet(value)) {
+                final long language = contentlet != null ? contentlet.getLanguageId() : languageAPI.getDefaultLanguage().getId();
+                relatedContentlets = filterContentlet(language, value, user);
+
+                if (contentlet != null){
```

Comment:
why not to include the
```
relatedContentlets = filterContentlet(language, value, user);
```
inside the 
```
if (contentlet != null){
```

Reply:
because the filterContentlet method always needs to be executed no matter if the contentlet is set or not